src/plugin/luapi_application.h

```diff
@@ -475,6 +476,194 @@ static int applib_changeBackgroundPdfPageNr(lua_State* L) {
+
+    std::string size = toolSizeToString(toolHandler->getPenSize());
+    int fillOpacity = toolHandler->getPenFill();
+    bool fill = toolHandler->getPenFillEnabled();
+
+    lua_newtable(L);
+
+    lua_pushliteral(L, "size");
+    lua_pushstring(L, size.c_str());
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fill");
+    lua_pushboolean(L, fill);
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fillOpacity");
+    lua_pushinteger(L, fillOpacity);
+    lua_settable(L, -3);
+
+    return 1;
+}
+
+/*
+ * Returns a table encoding all info on the highlighter in a Lua table of the shape
+ * {
+ *   "size" = string
+ *   "fill" = bool
+ *   "fillOpacity" = integer
+ * }
+ *
+ * See /src/control/ToolEnums.cpp for possible values of "size".
+ *
+ * Example 1: local highlighterInfo = app.getHighlighterInfo()
+ *            local size = highlighterInfo["size"]
+ *            local opacity = highlighterInfo["fillOpacity"]
+ *
+ * Example 2: local fill = app.getHighlighterInfo()["fill"]
+ */
+static int applib_getHighlighterInfo(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* control = plugin->getControl();
+    ToolHandler* toolHandler = control->getToolHandler();
+
+    std::string size = toolSizeToString(toolHandler->getHighlighterSize());
+    int fillOpacity = toolHandler->getHighlighterFill();
+    bool fill = toolHandler->getHighlighterFillEnabled();
+
+    lua_newtable(L);
+
+    lua_pushliteral(L, "size");
+    lua_pushstring(L, size.c_str());
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fill");
+    lua_pushboolean(L, fill);
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fillOpacity");
+    lua_pushinteger(L, fillOpacity);
+    lua_settable(L, -3);
+
+    return 1;
+}
+
+/*
+ * Returns a table encoding all info on the eraser in a Lua table of the shape
+ * {
+ *   "type" = string
+ *   "size" = string
+ * }
+ *
+ * See /src/control/ToolEnums.cpp for possible values of "type" and "size".
+ *
+ * Example 1: local eraserInfo = app.getEraserInfo()
+ *            local type = eraserInfo["type"]
+ *            local size = eraserInfo["size"]
+ *
+ * Example 2: local type = app.getEraserInfo()["type"]
+ */
+static int applib_getEraserInfo(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* control = plugin->getControl();
+    ToolHandler* toolHandler = control->getToolHandler();
+
+    std::string type = eraserTypeToString(toolHandler->getEraserType());
+    std::string size = toolSizeToString(toolHandler->getEraserSize());
+
+    lua_newtable(L);
+
+    lua_pushliteral(L, "type");
+    lua_pushstring(L, type.c_str());
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "size");
+    lua_pushstring(L, size.c_str());
+    lua_settable(L, -3);
+
+    return 1;
+}
+
+
+/*
+ * Returns a table encoding all info on the active tool in a Lua table of the shape
+ * {
+ *   "type" = string
+ *   "size" = string
+ *   "color" = integer
+ *   "fillOpacity" = integer
```

Comment:
From this line, it isn't clear that `fillOpacity` ranges from `0` to `255` (inclusive).

Additionally, it might make sense to make `fillOpacity` range from `0.0` to `1.0`, instead. This seems more intuitive (and more closely matches the GUI where `fillOpacity` ranges from `0.0` to `100.0`).

Reply:
If we break `color` up into `{ red = number, green = number, blue = number }`, we should also have each component have the same range as `fillOpacity`.

Reply:
I agree that it should be clarified that `fillOpacity` ranges from 0 to 255. I am not so happy with converting it to the 0.0 to 1.0 range when it is internally stored as an integer from 0 to 255. There may be rounding problems with the conversion.

Reply:
> There may be rounding problems with the conversion.

Okay! That makes sense!